addons/spreadsheet_dashboard/static/src/bundle/dashboard_action/dashboard_loader.js

```diff
@@ -134,13 +144,21 @@ export class DashboardLoader {
             {
                 specification: {
                     name: {},
-                    published_dashboard_ids: { fields: { name: {} } },
+                    published_dashboard_ids: { fields: { name: {}, is_favorite: {} } },
                 },
             }
         );
         return groups.records;
     }
 
+    /**
+     * Filters and returns an array of favorite dashboards.
+     * @returns {Array<Dashboard>}
+     */
+    _getFavoriteDashboards() {
+        return Object.values(this.dashboards).filter((dashboard) => dashboard.isFavorite);
```

Comment:
I would `.flatMap(...).filter(...)` on this `this.groups` instead. At least the order would be deterministic and understanble to users. Here it depends on the order in which the keys have been added to the `this.dashboards` object, which is not robust at all.

Reply:
@LucasLefevre, I’ve made the changes you suggested. I just want to point out that we also need to match the format of how the other dashboards are stored in the groups